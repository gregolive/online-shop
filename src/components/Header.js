import '../styles/Header.css';
import '@fortawesome/fontawesome-free/js/all';
import { Link } from "react-router-dom";

const Header = ({ openCart, cartCount }) => {
  return (
    <header className="SiteHeader">
      <Link to='/' className='HeaderLogo'>
        <img src={require('../img/logo.png')} alt='prospresso logo' />
        <h1>prospresso</h1>
      </Link>

      <div className='HeaderLinks'>
        <Link to='/'>Home</Link>
        <Link to='/shop'>Shop</Link>
        <button type='button' className='Button TransparentButton' onClick={openCart}>
          <i className='fa-solid fa-cart-shopping'></i>
          <span className='CartNum'>{cartCount}</span>
        </button>
      </div>
    </header>
  );
};

export default Header;
