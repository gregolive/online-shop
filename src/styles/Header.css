.SiteHeader {
  background: var(--primary);
  height: 5rem;
  padding: 0 6vw;
  display: grid;
  grid-template-columns: 1fr auto auto;
  gap: 4vw;
  align-items: center;
  justify-content: start;
}

.MenuButton {
  font-size: 1.2rem;
  padding: 0.5rem;
  display: none;
}

.HeaderLogo {
  color: var(--text);
  display: flex;
  align-items: center;
  gap: 0.75rem;
}

.HeaderLogo img {
  height: 3rem;
}

h1 {
  font-size: 2.2rem;
  font-weight: 700;
  margin: 0;
}

.HeaderLinks {
  color: var(--text);
  font-size: 1.1rem;
  display: flex;
  align-items: center;
  gap: 4vw;
}

.CartButton {
  font-size: 1.2rem;
  padding: 0.5rem 1rem 0.5rem 0;
  position: relative;
}

.CartNum {
  color: var(--primary);
  background: var(--highlight);
  font-size: 0.8rem;
  height: 1.25rem;
  aspect-ratio: 1;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0;
  right: 0;
}

@media screen and (max-width: 768px) {
  .SiteHeader {
    display: flex;
    justify-content: space-between;
  }

  .MenuButton {
    display: block;
  }

  .HeaderLinks {
    font-size: 1.3rem;
    background: transparent;
    visibility: hidden;
    opacity: 0;
    border-bottom: none;
    display: grid;
    grid-template-rows: 2rem 2rem;
    justify-content: center;
    gap: 1rem;
    padding: 0.5rem 0 1rem 0;
    width: 100vw;
    position: fixed;
    left: 0;
    top: 5rem;
    z-index: 1;
    transition: all 0.3s ease;
  }

  .HeaderLinks.Open {
    background: var(--primary);
    visibility: visible;
    opacity: 1;
    border-bottom: 1px solid var(--text);
  }

  .HeaderLinks > a {
    text-align: center;
  }
}

@media screen and (max-width: 480px) {
  .HeaderLogo {
    gap: 0.5rem;
  }

  .HeaderLogo img {
    height: 2.2rem;
  }
  
  h1 {
    font-size: 1.8rem;
  }
}