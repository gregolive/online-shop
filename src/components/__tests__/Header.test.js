import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Header';

test('renders logo', () => {
  render(<MemoryRouter><Header /></MemoryRouter>);
  expect(screen.getByAltText(/prospresso logo/)).toBeInTheDocument();
});

test('renders correct heading', () => {
  render(<MemoryRouter><Header /></MemoryRouter>);
  expect(screen.getByRole('heading').textContent).toMatch(/prospresso/);
});

test('renders correct links', () => {
  render(<MemoryRouter><Header /></MemoryRouter>);
  expect(screen.getAllByRole('link')[1].textContent).toMatch(/Home/);
  expect(screen.getAllByRole('link')[2].textContent).toMatch(/Shop/);
});

test('renders cart and menu buttons', () => {
  render(<MemoryRouter><Header /></MemoryRouter>);
  expect(screen.getAllByRole('button').length).toEqual(2);
});

it('renders consistently', () => {
  const { container } = render(<MemoryRouter><Header /></MemoryRouter>);
  expect(container).toMatchSnapshot();
});
