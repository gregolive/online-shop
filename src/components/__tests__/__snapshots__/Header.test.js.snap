// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistently 1`] = `
<div>
  <header
    class="SiteHeader"
  >
    <button
      class="Button TransparentButton MenuButton"
      type="button"
    >
      <i
        class="fa-solid fa-bars"
      />
    </button>
    <a
      class="HeaderLogo"
      href="/"
    >
      <img
        alt="prospresso logo"
        src="logo.png"
      />
      <h1>
        prospresso
      </h1>
    </a>
    <div
      class="HeaderLinks"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Shop
      </a>
    </div>
    <button
      class="Button TransparentButton CartButton"
      type="button"
    >
      <i
        class="fa-solid fa-cart-shopping"
      />
      <span
        class="CartNum"
      />
    </button>
  </header>
</div>
`;
